serving/src/main/java/feast/serving/specs/CachedSpecService.java

```diff
@@ -126,6 +139,19 @@ public FeatureSetSpec getFeatureSetSpec(String featureSetRef) throws ExecutionEx
                 FeatureSetSpec featureSetSpec = featureSetCache.get(fsName);
                 List<FeatureReference> requestedFeatures =
                     featureRefs.stream().map(Pair::getRight).collect(Collectors.toList());
+
+                // check that requested features reference point to different features in the
+                // featureset.
+                HashSet<String> featureNames = new HashSet<>();
```

Comment:
What is the downside of allowing this? (I am just thinking of the performance hit of keeping this in, even if its small).

Reply:
Its a limitation posed by this `FeatureSetRequest`:
https://github.com/feast-dev/feast/blob/14cae741068b4e18a56aaea211099564d78a703b/storage/api/src/main/java/feast/storage/api/retriever/FeatureSetRequest.java#L58